src/sentry/snuba/sessions_v2.py

```diff
@@ -341,16 +341,38 @@ def get_filter_conditions(self):
         conditions = SessionsV2QueryBuilder(**self.to_query_builder_dict()).where
         filter_conditions = []
         for condition in conditions:
-            # Exclude sessions "started" timestamp condition and org_id condition, as it is not needed for metrics queries.
+            if (unsupported_reason := self._is_supported_condition(condition)) is not None:
+                raise InvalidField(f"Invalid condition: {unsupported_reason}")
+
+            # Exclude sessions "started" timestamp condition and org_id condition, as it is not needed for metrics
+            # queries.
             if (
                 isinstance(condition, Condition)
                 and isinstance(condition.lhs, Column)
                 and condition.lhs.name in ["started", "org_id"]
             ):
                 continue
             filter_conditions.append(condition)
+
         return filter_conditions
 
+    @classmethod
+    def _is_supported_condition(cls, condition):
+        if isinstance(condition, BooleanCondition):
+            for nested_condition in condition.conditions:
+                if (
+                    unsupported_reason := cls._is_supported_condition(nested_condition)
+                ) is not None:
+                    return unsupported_reason
+        elif isinstance(condition, Condition):
+            if isinstance(condition.lhs, Function):
+                # Since we moved to metrics backed sessions, we don't allow wildcard search anymore. The reason for this
+                # is that we don't store tag values as strings in the database, this makes wildcard match on the
+                # db not possible. The solution would be to lift it out at the application level, but it will impact
+                # performance.
+                if condition.lhs.function == "match":
+                    return "wildcard search is not supported"
```

Comment:
nit: I would simplify this function to return nothing (call it `_validate_condition`), and just raise `InvalidField` on this line.

Reply:
I was thinking about this, it will make it more streamlined, I just wanted to scope it down to one exception type but yeah it doesn't look good.